src/main/java/org/shredzone/commons/suncalc/MoonTimes.java

```diff
@@ -52,23 +54,15 @@ public static MoonTimes of(Date date, double lat, double lng) {
      * Calculates the {@link MoonTimes}, based on the given {@link TimeZone}.
      */
     public static MoonTimes of(Date date, double lat, double lng, TimeZone tz) {
-        Calendar cal = Calendar.getInstance(tz);
-        cal.setTime(date);
-        cal.set(Calendar.HOUR_OF_DAY, 0);
-        cal.set(Calendar.MINUTE, 0);
-        cal.set(Calendar.SECOND, 0);
-        cal.set(Calendar.MILLISECOND, 0);
-        Date t = cal.getTime();
-
         double hc = 0.133 * RAD;
-        double h0 = MoonPosition.of(t, lat, lng).getAltitude() - hc;
+        double h0 = MoonCalculationsUtil.preciseAltitude(date, tz, 0, lat, lng) - hc;
         Double rise = null, set = null;
```

Comment:
not really, make it a little easier to read, I noticed that it was truncating before and now you aren't was curious if that helped with the calculations or if you were truncating later and I missed it.